File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        int l=0;
        HashSet<Integer> set = new HashSet<>();
        for(int r=0, n=nums.length; r<n; ++r){
            set.add(nums[r]);
            if(r-l == k){
                set.remove(nums[l]);
                l+=1;
            }
            if(set.size() != r-l+1)
                return true;
        }
        return false;
    }
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        l=0
        s=set()
        for r in range(0, len(nums)):
            s.add(nums[r])
            if r-l+1 > k:
                s.remove(nums[l])
                l+=1
            if len(s)!=r-l+1:
                return True
        return False